xyz-hub-service/src/main/java/com/here/xyz/hub/rest/BranchApi.java

```diff
@@ -52,8 +52,16 @@ private Future<List<Branch>> getBranches(RoutingContext context) {
   }
 
   private Future<Branch> postBranch(RoutingContext context) throws HttpException {
-    return BranchHandler.createBranch(getMarker(context), getSpaceId(context), getBranchPayload(context))
-        .recover(t -> Future.failedFuture(new HttpException(BAD_REQUEST, "Invalid request body")));
+    String spaceId = getSpaceId(context);
+    Branch branch = getBranchPayload(context);
+    return checkExistingAlias(getMarker(context), spaceId, branch.getId())
+        .compose(v -> BranchHandler.createBranch(getMarker(context), getSpaceId(context), branch))
+        .recover(t -> {
+          if (t instanceof HttpException)
+            return Future.failedFuture(t);
+          else
```

Comment:
All exceptions other than HttpExceptions would be transformed into HTTP 400s  in that case. Is that really intended?